File: LeetCode Solutions/March 2022 LeetCoding Challenges/Is Subsequence/Is Subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int s_i = 0, t_i = 0;
        while (s_i < s.size() && t_i < t.size()){
            if (s[s_i] == t[t_i]){
                s_i++;
            }
            t_i++;
        }
        return s_i == s.size();
    }
};

File: LeetCode Solutions/March 2022 LeetCoding Challenge/Is Subsequence/Is Subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int s_i = 0, t_i = 0;
        while (s_i < s.length() && t_i < t.length()){
            if (s.charAt(s_i) == t.charAt(t_i)){
                s_i++;
            }
            t_i++;
        }
        return s_i == s.length();
    }
}